File: 002.py
def func(thresh):
    a = 1
    b = 1
    ret = 0
    while 1:
        a, b = a + b, a
        if a > thresh:
            return ret
        if a % 2 == 0:
            ret += a

print func(4*1000*1000)

File: 002.clj
(defn fib [thresh]
  (defn gen [lst]
    (let [n (+ (nth lst 0) (nth lst 1))]
      (if (> n thresh) lst
        (gen (cons n lst)))))
  (gen '(2 1)))

(println (apply + (filter even? (fib (* 4 1000 1000)))))
